frontend/app/configuration/Config.scala

```diff
@@ -35,8 +36,15 @@ object Config {
 
   val idWebAppUrl = config.getString("identity.webapp.url")
 
-  def idWebAppSigninUrl(uri: String): String =
-    (idWebAppUrl / "signin") ? ("returnUrl" -> s"$membershipUrl$uri") ? ("skipConfirmation" -> "true")
+  def idWebAppSigninUrl(uri: String): String = {
+    val encodedUri = URLEncoder.encode(membershipUrl + uri, "UTF-8")
+    (idWebAppUrl / "signin") ? ("returnUrl" -> s"$encodedUri") ? ("skipConfirmation" -> "true")
+  }
+
+  def idWebAppSigninUrlExternal(uri: String): String = {
```

Comment:
This is a little awkward but `idWebAppSigninUrl` hard appends the membership domain which doesn't help when trying to return a user to an external domain. Needs to be refactored to be one method at some point